models/baseModels/InvoiceItem/InvoiceItem.ts

```diff
@@ -636,7 +637,40 @@ export abstract class InvoiceItem extends Doc {
         itemNotFor = 'Purchases';
       }
 
-      return { for: ['not in', [itemNotFor]] };
+      const baseFilter: QueryFilter = {
+        for: ['not in', [itemNotFor]],
+      };
+
+      const accountingSettings = doc.fyo.singles?.AccountingSettings;
+      const inventorySettings = doc.fyo.singles?.InventorySettings;
+
+      if (accountingSettings?.enablePointOfSaleWithOutInventory) {
+        const posSettings = doc.fyo.singles?.POSSettings;
+
+        const posItemVisibility = posSettings?.itemVisibility;
+
+        if (posItemVisibility === 'Inventory Items') {
+          return { ...baseFilter, trackItem: true };
+        }
+
+        if (posItemVisibility === 'Non-Inventory Items') {
+          return { ...baseFilter, trackItem: false };
+        }
+
+        return baseFilter;
+      }
+
+      const invItemVisibility = inventorySettings?.itemVisibility;
+
+      if (invItemVisibility === 'Inventory Items') {
+        return { ...baseFilter, trackItem: true };
+      }
+
+      if (invItemVisibility === 'Non-Inventory Items') {
+        return { ...baseFilter, trackItem: false };
+      }
+
+      return baseFilter;
```

Comment:
We could get better readability if `invItemVisibility` was a boolean field @Gadha2311